docs/Getting_Started.md

```diff
@@ -0,0 +1,421 @@
+
+### Adding permissions
+
+Apps need approval to perform actions inside of Discord (like installing a slash command or adding emojis). So before installing your app, let's add some permissions. Click on **OAuth2** in the left sidebar, then **URL generator**.
+
+> info
+> The URL generator helps create an installation link with the permissions your app needs to function. You can use the link to install the app onto your server, or share it with others so they can install it on their own.
+
+For now, we’ll just add a couple of permissions:
+- `applications.commands` lets your app create commands in guilds its installed
+- `bot` is to enable your bot user. After you click bot, you can also add individual permissions. For now, just check **Send Messages**.
+
+![URL generator screenshot](url-generator.png)
+
+Once you add permissions, you should see a URL that you can copy to install your app.
+
+### Installing your app
+
+Copy the URL from above, and paste it into your browser. You’ll be guided through the installation flow, where you should make sure you’re installing the app on a server where you can develop and test.
+
+After installing your app, you can head over to your server and see that it has joined ✨
+
+With your app configured and installed, let’s start developing it.
+
+## Running your app
+
+All of the code used in the example app can be found in [the Github repository](https://github.com/shaydewael/discord-getting-started).
+
+To make development a bit simpler, the app uses [discord-interactions](https://github.com/discord/discord-interactions-js), which provides types and helper functions. If you prefer to use other languages or libraries, there’s [a page with community-built resources](#DOCS_TOPICS_COMMUNITY_RESOURCES) which you can browse through.
+
+### Remix the project
+
+This guide uses Glitch, which allows you to quickly clone and develop an app from within your browser. There are also instructions on developing locally using ngrok [in the README](https://github.com/shaydewael/discord-getting-started#running-app-locally) if you'd prefer.
+
+> info
+> While Glitch is great for development and testing, [it has technical limitations](https://help.glitch.com/kb/article/17-technical-restrictions/) so other hosting providers should be considered for production apps.
+
+To start, **[remix (or clone) the Glitch project 🎏](https://glitch.com/edit/#!/remix/getting-started-discord)**
+
+When you remix the project, you'll see a new Glitch project with a unique name similar to this:
+
+![Glitch project overview](glitch-project.png)
+
+#### Project structure
+
+All of the files for the project are on the left-hand side. Here's a quick glimpse at the structure:
+
+```
+├── examples    -> short, feature-specific sample apps
+│   ├── app.js  -> completed app.js code
+│   ├── button.js
+│   ├── command.js
+│   ├── modal.js
+│   ├── selectMenu.js
+├── .env        -> your credentials and IDs
```

Comment:
I have some concerns over using the dotenv approach to storing secrets.  Invariably, people will check these things into source control, or push them to production where they end up getting otherwise exposed.  I like the simplicity, but it's worth asking if we should consider another approach to storing secrets for development.

Reply:
Isn't it ignored in .gitignore?